utils/s2n_safety.c

```diff
@@ -134,8 +143,21 @@ int s2n_constant_time_pkcs1_unpad_or_dont(uint8_t * dst, const uint8_t * src, ui
 
     dont_copy |= src[0] ^ 0x00;
     dont_copy |= src[1] ^ 0x02;
-    dont_copy |= start_of_data[-1] ^ 0x00;
 
+/* Since -1 is being used, we need to disable the pointer overflow check for CBMC. */
+#pragma CPROVER check push
+#pragma CPROVER check disable "pointer-overflow"
+    dont_copy |= start_of_data[-1] ^ 0x00;
```

Comment:
Are we certain this is legal C?

Reply:
I looked it up and it appears to be allowed, since it's just an addition to a pointer address.

Reply:
@camshaft any suggestions here too?